### PR TITLE
Core/Unit: Implement more helper methods

### PR DESCRIPTION
**Pull request by ForesterDev**
**Changes proposed:**

-  Add more helper methods for Unit and Player classes

**Target branch(es):** 3.3.5/master

- [x] 3.3.5

### CODE REVIEW

**Comment by jackpoz** [comment]
this PR is growing quite much as it tries to do a lot of things at the same time. Maybe 1 PR for every renamed method would allow to focus better on the changes.

**Comment by ForesterDev** [comment]
@jackpoz ok, i'he reverted last changes and will create another PR when this PR will be merged :)

File: src/server/game/Entities/Player/Player.h [hunk 1]
```diff
@@ -967,6 +967,8 @@ class TC_GAME_API Player : public Unit, public GridObject<Player>
         void SetPvPDeath(bool on) { if (on) m_ExtraFlags |= PLAYER_EXTRA_PVP_DEATH; else m_ExtraFlags &= ~PLAYER_EXTRA_PVP_DEATH; }
 
         void GiveXP(uint32 xp, Unit* victim, float group_rate=1.0f);
+        uint32 GetPlayerXP() const { return GetUInt32Value(PLAYER_XP); }
+        void SetPlayerXP(uint32 newXP) { SetUInt32Value(PLAYER_XP, newXP); }
```
Comment (Shauren) [inline src/server/game/Entities/Player/Player.h]: This already exists in master branch and is named `SetXP`, some consistency would be great (even if it did not exist, the field is named `XP`, not `PLAYER_XP`, `PLAYER_` is just a prefix for enum value)

**Comment by Wyrserth** [comment]
Looks good to me :)

**Comment by ForesterDev** [comment]
It's ready for merge

**Comment by Wyrserth** [comment]
Thanks for your PR!

**Pull request merged**